design/ceph/object/store.md

```diff
@@ -351,3 +351,212 @@ spec:
+
+Ceph administrator creates set of `index_pool`, `data_pool`, and `data_extra_pool` per each target placement. Select arbitrary names for target placements. For example `hot` and `cold`:
+```yaml
+# Bucket Data pool for hot storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: hot.rgw.buckets.data
+  namespace: rook-ceph 
+spec:
+  ...
+---
+# Bucket index pool for hot storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: hot.rgw.buckets.index
+  namespace: rook-ceph
+spec:
+  ...
+---
+# Bucket data extra pool for hot storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: hot.rgw.buckets.non-ec
+  namespace: rook-ceph
+spec:
+  ...
+---
+# Bucket Data pool for cold storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: cold.rgw.buckets.data
+  namespace: rook-ceph 
+spec:
+  ...
+---
+# Bucket index pool for cold storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: cold.rgw.buckets.index
+  namespace: rook-ceph
+spec:
+  ...
+---
+# Bucket data extra pool for cold storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: cold.rgw.buckets.non-ec
+  namespace: rook-ceph 
+spec:
+  ...
+```
+If needed, create additional pool to use it in custom StorageClass, for example `GLACIER`:
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: glacier.rgw.buckets.data
+  namespace: rook-ceph 
+spec:
+  ...
+```
+Then update `CephObjectStore` CRD with target placements and StorageClasses definitions:
+```yaml
+spec:
+  targetPlacements:                         # [optional] list of target placements
+  # target placement named hot:
+  - name: hot                               # [required] should be unique
+    dataPool: hot.rgw.buckets.data          # [required] default data pool, will be used with Ceph default STANDARD StorageClass, can be overridden with custom StorageClasses
+    indexPool: hot.rgw.buckets.index        # [required] 
+    dataExtraPool: hot.rgw.buckets.non-ec   # [required] 
+    default: false                          # true|false only one target must be set as default
+  # target placement named cold
+  - name: cold                              # [required] should be unique
+    dataPool: cold.rgw.buckets.data         # [required] default data pool, will be used with Ceph default STANDARD StorageClass, can be overridden with custom StorageClasses
+    indexPool: cold.rgw.buckets.index       # [required] 
+    dataExtraPool: cold.rgw.buckets.non-ec  # [required] 
+    default: true                           # true|false only one target must be set as default
```

Comment:
The thing that I am most curious/worried about is what support should look like if different placements want to share certain pools. Like, what if a user wants all placements to use the same index pools? Is this supported? Is it not? Does it require namespaced pools to support?

@cbodley do you have any guidance here?

Reply:
it's fine for placement targets/storage classes to share pools as long as they use different namespaces

Reply:
@arttor let's be sure to note this here and for users in eventual docs.

IMO, it would be best to provide examples for users of shared pools, because that scenario is the more complicated one for users. IMO, it would be best if we focused development on that combo as well, because if shared pools works, non-shared pools should automatically work, but the reverse is not guaranteed.